packages/gatsby-cli/src/reporter/reporter-progress.ts

```diff
@@ -0,0 +1,109 @@
+import reporterActions from "./redux/actions"
+import { ActivityStatuses, ActivityTypes } from "./constants"
+import { Span } from "opentracing"
+import { reporter as gatsbyReporter } from "./reporter"
+
+export interface ICreateProgressReporterArguments {
+  id: string
+  text: string
+  start: number
+  total: number
+  span: Span
+  reporter: typeof gatsbyReporter
+}
+
+export type ProgressReporter = ReturnType<typeof createProgressReporter>
+
+export const createProgressReporter = ({
+  id,
+  text,
+  start,
+  total,
+  span,
+  reporter,
+}: ICreateProgressReporterArguments) => {
```

Comment:
Add a comment and lint exception explaining why you're not typing the return value